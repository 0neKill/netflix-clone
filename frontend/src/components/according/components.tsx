import React from 'react';
import {Container, Inner, Item, Header, Title, Body} from './styles';


interface IToggleContext {
    selectedItem: number | null,
    handlerSetToggle: (id: number) => void,
}

const ToggleContext = React.createContext<IToggleContext>({} as IToggleContext);


export default function According({children}: React.PropsWithChildren<unknown>) {
    const [selectedItem, setSelectedItem] = React.useState<number | null>(null);

    const handlerSetToggle = (id: number) => {
        setSelectedItem(prevId => prevId === id ? null : id);
    }

    return (
        <Container>
            <ToggleContext.Provider value={{selectedItem, handlerSetToggle}}>
                <Inner>{children}</Inner>
            </ToggleContext.Provider>
        </Container>
    )
}


According.Title = function AccordingTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
}


According.Item = function AccordingItem({children}: React.PropsWithChildren<unknown>) {
    return <Item>{children}</Item>
}

According.Header = function AccordingHeader({children, id}: React.PropsWithChildren<{ id: number }>) {
    const {handlerSetToggle, selectedItem} = React.useContext(ToggleContext);
    return <Header onClick={handlerSetToggle.bind(this, id)}>
        {children}
        {
            selectedItem === id ?
                <img src='/images/icons/close-slim.png' alt='Close'/> :
                <img src='/images/icons/add.png' alt='Open'/>
        }
    </Header>
}
According.Body = function AccordingBody({children, id}: React.PropsWithChildren<{ id: number }>) {
    const {selectedItem} = React.useContext(ToggleContext);
    return selectedItem === id ? <Body>{children}</Body> : null;
}