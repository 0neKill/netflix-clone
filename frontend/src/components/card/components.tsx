import React from "react";

import {
    Container,
    Group,
    Title,
    SubTitle,
    Text,
    Meta,
    Item,
    Image,
    Entities,
    Feature,
    FeatureClose,
    FeatureText,
    FeatureTitle,
    Content,
    Maturity
} from './styles';

import {ICardItem, ItemType, IFeatureCardContext, IImage} from "./@types";


const FeatureCardContext = React.createContext<IFeatureCardContext>({} as IFeatureCardContext);

export default function Card({children}: React.PropsWithChildren<unknown>) {
    const [itemFeature, setItemFeature] = React.useState<ItemType>({} as ItemType);
    const [isShowFeature, setIsShowFeature] = React.useState<boolean>(false);
    return (
        <FeatureCardContext.Provider value={{
            itemFeature,
            isShowFeature,
            setItemFeature,
            setIsShowFeature
        }}>
            <Container>{children}</Container>
        </FeatureCardContext.Provider>
    )
}

Card.Group = function CardGroup({children}: React.PropsWithChildren<unknown>) {
    return <Group>{children}</Group>
}
Card.Title = function CardTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
}
Card.SubTitle = function CardSubTitle({children}: React.PropsWithChildren<unknown>) {
    return <SubTitle>{children}</SubTitle>
}
Card.Text = function CardText({children}: React.PropsWithChildren<unknown>) {
    return <Text>{children}</Text>
}

Card.Entities = function CardEntities({children}: React.PropsWithChildren<unknown>) {
    return <Entities>{children}</Entities>
}

Card.Meta = function CardMeta({children}: React.PropsWithChildren<unknown>) {
    return <Meta>{children}</Meta>
}

Card.Feature = function CardFeature({children}: React.PropsWithChildren<unknown>) {
    const {isShowFeature, itemFeature, setIsShowFeature} = React.useContext<IFeatureCardContext>(FeatureCardContext);
    console.log(isShowFeature)
    return isShowFeature ?
        <Feature src={itemFeature.images.large}>
            <Content>
                <FeatureTitle>{itemFeature.title}</FeatureTitle>
                <FeatureText>{itemFeature.description}</FeatureText>
                <FeatureClose onClick={() => setIsShowFeature(false)}>
                    <img src='/images/icons/close.png' alt='close'/>
                </FeatureClose>
                <Group margin='30px 0' flexDirection='row' alignItems='center'>
                    <Maturity rating={itemFeature.maturity}>
                        {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
                    </Maturity>
                    <FeatureText fontWeight='bold'>
                        {
                            itemFeature.genre.charAt(0).toUpperCase() + itemFeature.genre.slice(1)
                        }
                    </FeatureText>
                </Group>
                {children}
            </Content>
        </Feature> : null;
}

Card.Item = function CardItem({item, children}: React.PropsWithChildren<ICardItem>) {
    const {setItemFeature, setIsShowFeature} = React.useContext<IFeatureCardContext>(FeatureCardContext);
    const handlerOnClick = () => {
        console.log('Item')
        setItemFeature(item);
        setIsShowFeature(true);
    }
    return <Item onClick={handlerOnClick}>{children}</Item>
}

Card.Image = function CardImage({src, alt}: React.PropsWithChildren<IImage>) {
    return <Image src={src} alt={alt}/>
}
