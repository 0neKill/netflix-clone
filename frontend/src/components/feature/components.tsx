import React from "react";

import {Container, Title, SubTitle} from './styles';

export default function Feature({children}: React.PropsWithChildren<unknown>) {
    return <Container>{children}</Container>
}

Feature.Title = function FeatureTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
}
Feature.SubTitle = function FeatureSubTitle({children}: React.PropsWithChildren<unknown>) {
    return <SubTitle>{children}</SubTitle>
}