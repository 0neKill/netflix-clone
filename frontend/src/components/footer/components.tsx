import React from "react";
import {ILink, } from "./@types";
import {Container, Row, Link, Text, Title, Column, Break} from "./styles";


export default function Footer({children}: React.PropsWithChildren<unknown>) {
    return <Container>{children}</Container>
}


Footer.Row = function FooterRow({children}: React.PropsWithChildren<unknown>) {
    return <Row>{children}</Row>
}


Footer.Column = function FooterColumn({children}: React.PropsWithChildren<unknown>) {
    return <Column>{children}</Column>
}

Footer.Link = function FooterLink({children, href}: React.PropsWithChildren<ILink>) {
    return <Link href={href}>{children}</Link>
}

Footer.Title = function FooterTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
}

Footer.Text = function FooterText({children}: React.PropsWithChildren<unknown>) {
    return <Text>{children}</Text>
}
Footer.Break = function FooterBreak({children}: React.PropsWithChildren<unknown>) {
    return <Break>{children}</Break>
}