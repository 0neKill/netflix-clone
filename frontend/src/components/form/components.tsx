import React from "react";

import {Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit} from './styles';
import {IBase, ILink, ISubmit, ITextInput} from "./@types";


export default function Form({children}: React.PropsWithChildren<unknown>) {
    return <Container>{children}</Container>
}

Form.Base = function FormBase({children, handlerSubmit}: React.PropsWithChildren<IBase>) {
    return <Base onSubmit={handlerSubmit}>{children}</Base>
}
Form.Error = function FormError({children}: React.PropsWithChildren<unknown>) {
    return <Error>{children}</Error>
}
Form.Title = React.memo(function FormTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
})
Form.Text = function FormText({children}: React.PropsWithChildren<unknown>) {
    return <Text>{children}</Text>
}
Form.TextSmall = React.memo(function FormTextSmall({children}: React.PropsWithChildren<unknown>) {
    return <TextSmall>{children}</TextSmall>
})
Form.TextLink = function FormLink({children, to}: React.PropsWithChildren<ILink>) {
    return <Link to={to}>{children}</Link>
}
Form.TextInput = React.memo(function FormInput({
                                                   children,
                                                   type,
                                                   onChange,
                                                   value,
                                                   placeholder
                                               }: React.PropsWithChildren<ITextInput>) {
    return <Input placeholder={placeholder} type={type} onChange={onChange} value={value}>{children}</Input>
})
Form.Submit = function FormSubmit({children, type, disable}: React.PropsWithChildren<ISubmit>) {
    return <Submit type={type} disabled={disable}>{children}</Submit>
}
