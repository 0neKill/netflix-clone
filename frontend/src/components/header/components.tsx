import React from "react";
import {Link} from 'react-router-dom';
import {ILogo, IPicture, IProps, IRoutes, ITextLink, ISearch} from "./@types";
import {
    Background,
    Logo,
    Container,
    ButtonLink,
    Feature,
    Text,
    FeatureCallOut,
    TextLink,
    Group,
    Picture,
    Profile,
    Dropdown,
    Search,
    SearchIcon,
    SearchInput,
    PlayButton
} from "./styles";


export default function Header({bg = true, src, dontShowOnSmallViewPort, children}: React.PropsWithChildren<IProps>) {
    return bg ? <Background src={src}
                            dontShowOnSmallViewPort={dontShowOnSmallViewPort}>{children}</Background> : <>{children}</>;
}

Header.Frame = function HeaderFrame({children}: React.PropsWithChildren<unknown>) {
    return <Container>{children}</Container>
}
Header.Search = function HeaderSearch({searchTerm, handlerChangeSearchTerm}: React.PropsWithChildren<ISearch>) {
    const [isActive, setIsActive] = React.useState<boolean>(false);

    const handlerIsActiveSearch = () => setIsActive(isActive => !isActive);
    console.log(isActive)
    return (
        <Search>
            <SearchIcon onClick={handlerIsActiveSearch}>
                <img src='/images/icons/search.png' alt='Search'/>
            </SearchIcon>
            <SearchInput value={searchTerm}
                         onChange={e => handlerChangeSearchTerm(e.target.value)}
                         placeholder='Search films and series'
                         active={isActive}
            />
        </Search>
    )
}
Header.Group = function HeaderGroup({children}: React.PropsWithChildren<unknown>) {
    return <Group>{children}</Group>
}
Header.Profile = function HeaderProfile({children}: React.PropsWithChildren<unknown>) {
    return <Profile>{children}</Profile>
}
Header.Dropdown = function HeaderDropdown({children}: React.PropsWithChildren<unknown>) {
    return <Dropdown>{children}</Dropdown>
}

Header.Picture = function HeaderPicture({src}: React.PropsWithChildren<IPicture>) {
    return <Picture src={src}/>
}

Header.Feature = function HeaderFeature({children}: React.PropsWithChildren<unknown>) {
    return <Feature>{children}</Feature>
}
Header.FeatureCallOut = function HeaderFeatureCallOut({children}: React.PropsWithChildren<unknown>) {
    return <FeatureCallOut>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({children}: React.PropsWithChildren<unknown>) {
    return <Text>{children}</Text>
}
Header.TextLink = function HeaderTextLink({children, active}: React.PropsWithChildren<ITextLink>) {
    return <TextLink active={active}>{children}</TextLink>
}

Header.Logo = function HeaderLogo({to, alt, src}: React.PropsWithChildren<ILogo>) {
    return (
        <Link to={to}>
            <Logo src={src} alt={alt}/>
        </Link>
    )
}
Header.ButtonLink = function HeaderButtonLink({children, to}: React.PropsWithChildren<IRoutes>) {
    return <ButtonLink to={to}>{children}</ButtonLink>
}
Header.PlayButton = function HeaderPlayButton({children}: React.PropsWithChildren<unknown>) {
    return <PlayButton>{children}</PlayButton>
}
