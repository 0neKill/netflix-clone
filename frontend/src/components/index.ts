export {JumbotronContainer} from './jumbotron';
export {FooterContainer} from './footer';
export {AccordingContainer} from './according';
export {OptFormContainer} from './opt-form';
export {HeaderContainer} from './header';
export {FeatureContainer} from './feature';
export {ProfileContainer} from './profile';
export {LoadingContainer} from './loading';
export {CardContainer} from './card';
export {PlayerContainer} from './player';