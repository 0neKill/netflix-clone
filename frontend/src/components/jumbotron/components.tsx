import React from "react";

import {Container, Inner, Pane, Title, SubTitle, Image, Item} from './styles';

import {IImage, IProps} from "./@types";

export default function Jumbotron({children, direction = 'row'}: React.PropsWithChildren<IProps>) {
    return (
        <Item>
            <Inner direction={direction}>
                {children}
            </Inner>
        </Item>

    )
}


Jumbotron.Container = function JumbotronContainer({children}: React.PropsWithChildren<unknown>) {
    return <Container>{children}</Container>
}

Jumbotron.Pane = function JumbotronPane({children}: React.PropsWithChildren<unknown>) {
    return <Pane>{children}</Pane>
}

Jumbotron.Title = function JumbotronTitle({children}: React.PropsWithChildren<unknown>) {
    return <Title>{children}</Title>
}

Jumbotron.SubTitle = function JumbotronSubTitle({children}: React.PropsWithChildren<unknown>) {
    return <SubTitle>{children}</SubTitle>
}


Jumbotron.Image = function JumbotronImage({children, ...restProps}: React.PropsWithChildren<IImage>) {
    return <Image src={restProps.src} alt={restProps.alt}>{children}</Image>
}



