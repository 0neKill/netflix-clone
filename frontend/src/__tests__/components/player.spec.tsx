import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import useEvent from '@testing-library/user-event';
import {PlayerContainer} from '../../components';


describe('Player component', () => {
    it('render the Player component', () => {
        const {container} = render(<PlayerContainer/>);
        const playerButton = screen.getByText(/Play/i);
        expect(screen.queryByTestId('player')).toBeNull();
        fireEvent.click(playerButton);
        expect(container).toMatchSnapshot();
        expect(screen.getByTestId('player')).not.toBeNull();
    })
})

