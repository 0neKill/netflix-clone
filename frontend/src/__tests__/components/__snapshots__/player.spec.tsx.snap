// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component render the Player component 1`] = `
<div>
  <div
    class="sc-faUpoM"
  >
    <button
      class="sc-TBWPX jkvZCf"
    >
      Play
    </button>
  </div>
</div>
`;
